package.json

```diff
@@ -24,26 +24,22 @@
   ],
   "scripts": {
     "build": "tsc",
+    "check": "tsc --noEmit",
+    "clean": "git clean -fXd",
     "test": "jest",
     "docs": "typedoc"
   },
-  "jest": {
-    "testEnvironment": "jsdom",
-    "transform": {
-      "\\.ts$": "ts-jest"
-    }
-  },
   "devDependencies": {
-    "@types/jest": "29.0.1",
-    "jest": "29.0.3",
-    "jest-environment-jsdom": "29.0.3",
-    "jest-fetch-mock": "3.0.3",
-    "ts-jest": "29.0.0",
-    "typedoc": "0.23.11",
-    "typescript": "4.8.3"
+    "@types/jest": "^29.0.2",
+    "@types/url-template": "^2.0.28",
+    "jest": "^29.0.3",
+    "jest-environment-jsdom": "^29.0.3",
+    "jest-fetch-mock": "^3.0.3",
+    "ts-jest": "^29.0.1",
+    "typedoc": "^0.23.14",
+    "typescript": "^4.8.3"
```

Comment:
I had previously pinned exact versions for dev dependencies, because Renovate keeps the up to date and this way builds are more reliably reproducible.

Reply:
Regarding the mixing: Currently all `devDependencies` are pinned in `package.json` and none of the regular `dependencies` are pinned.

Regarding pinning the `devDependencies` (in addition to `package-lock.json`): I was following [the reasoning from the Renovate docs](https://docs.renovatebot.com/dependency-pinning/).